### PR TITLE
+cuelang

### PR DESCRIPTION
**Pull request by johnallen3d**
Add `cue` from cuelang.org

### CODE REVIEW

File: projects/cuelang.org/package.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+build:
+  script: |
+    go mod download
+    go build -v -ldflags="$LDFLAGS" "{{ prefix }}"/bin/cue ./cmd/cue
+  dependencies:
+    go.dev: ^1.18
+  env:
+    LDFLAGS:
+      - -s -w -X cuelang.org/go/cmd/cue/cmd.version=v{{ version }}
```
Comment (johnallen3d) [inline projects/cuelang.org/package.yml]: cargo-culted from [brew](https://github.com/Homebrew/homebrew-core/blob/2b46e85bba6ce4615d13d4f018e938c1c623315a/Formula/cue.rb#L23).

File: projects/cuelang.org/package.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+versions:
+  github: cue-lange/cue
+
+provides:
+  - bin/cue
+
+build:
+  script: |
+    go mod download
+    go build -v -ldflags="$LDFLAGS" -o "{{ prefix }}"/bin/cue ./cmd/cue
```
Comment (johnallen3d) [inline projects/cuelang.org/package.yml:14]: There is directory in the root of this repo named `cue`. If I try to build locally without specifying an output file I receive the following error.

```sh
go: build output "cue" already exists and is a directory
```

**Comment by johnallen3d** [comment]
I've followed the contributing instructions but am getting an error when running the build script locally.

```sh
❯ ls -l
cli
pantry.core
pantry.extra

❯ pantry.core/scripts/build.ts cuelang.org
error: Uncaught Error:     https://github.com/teaxyz/pantry.zero#contributing

  throw new TeaError('not-found: pantry: package.yml', {project}, )
        ^
    at entry (https://raw.githubusercontent.com/teaxyz/cli/v0.18.1/src/hooks/usePantry.ts:218:9)
    at Object.getDeps (https://raw.githubusercontent.com/teaxyz/cli/v0.18.1/src/hooks/usePantry.ts:59:21)
    at file:///Users/john.allen/dev/src/forks/tea/pantry.core/scripts/build.ts:34:28
```

**Comment by jhheider** [comment]
does your `TEA_PANTRY_PATH` include your pantry.extra dir? that's how it picks up local changes.

File: projects/cuelang.org/package.yml (hunk 2 above)
Comment (jhheider) [reply projects/cuelang.org/package.yml:14]: i fixed a bug in the self-hosted (aarch64) runners caused by go mod leaving unwritable files behind, which prevented cleanup; hopefully this is fixed now.

**Comment by johnallen3d** [comment]
Yes @jhheider, I think I've set that correctly.

```sh
❯ pwd
/Users/john.allen/dev/src/forks/tea

❯ echo $TEA_PANTRY_PATH
/Users/john.allen/dev/src/forks/tea/pantry.extra
```

**Comment by jhheider** [comment]
Then it _should_ be working. Does it make any difference if you `cd pantry.extra` first? That'd likely be a bug, but such things aren't unheard of.

**Comment by johnallen3d** [comment]
Same result if I try to `cd pantry.extra` first @jhheider. Perhaps my local issues are beside the point? It looks like the last push I made (rebase from `main`) succeeded in several of the builds. The subsequent failures reveal an invalid S3 URL:

https://github.com/teaxyz/pantry.extra/actions/runs/3843213088/jobs/6545321988#step:7:107

**Comment by johnallen3d** [comment]
OK, I was able to get a successful local build. I'm a fish shell user but I decided to give bash a try and the build succeeded after switching shells. 🎉

**Comment by jhheider** [comment]
Yes... your build completed successfully, but github actions doesn't propagate secrets to action runs by non-organization/team members (for good security reasons, but it's frustrating for building any kind of secure pipeline for reusable artifacts). I've ticketed it here for tracking: https://github.com/teaxyz/pantry.core/issues/86.

That said, your work appears to pass the other checks, which makes it a candidate to merge. Good job!

File: projects/cuelang.org/package.yml (hunk 1 above)
Comment (jhheider) [reply projects/cuelang.org/package.yml]: recommend splitting these up for better legibility and audits (and using the long-form flags if available)

File: projects/cuelang.org/package.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    LDFLAGS:
+      - -s -w -X cuelang.org/go/cmd/cue/cmd.version=v{{ version }}
+    linux:
+      # or segmentation fault
+      # fix found here https://github.com/docker-library/golang/issues/402#issuecomment-982204575
+      LDFLAGS:
+        - -buildmode=pie
+
+test:
+  cue version
```
Comment (jhheider) [inline projects/cuelang.org/package.yml]: Can we write a better test that uses a `.cue` file to do something?

**Review by jhheider — COMMENTED** [review]
Looks good. Couple of style suggestions.

File: projects/cuelang.org/package.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  script: |
+    go mod download
+    go build -v -ldflags="$LDFLAGS" -o "{{ prefix }}"/bin/cue ./cmd/cue
+  dependencies:
+    go.dev: ^1.18
+  env:
+    LDFLAGS:
+      - -s
+      - -w
+      - -X cuelang.org/go/cmd/cue/cmd.version=v{{ version }}
```
Comment (johnallen3d) [inline projects/cuelang.org/package.yml:21]: @jhheider - I've checked the [documentation on link flags](https://pkg.go.dev/cmd/link) and it doesn't look like there are long-form options.

File: projects/cuelang.org/package.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    linux:
+      # or segmentation fault
+      # fix found here https://github.com/docker-library/golang/issues/402#issuecomment-982204575
+      LDFLAGS:
+        - -buildmode=pie
+
+test:
+  script: |
+    out=$(cue export data.cue --out yaml)
+    test "$out" = "project: cuelang.org"
```
Comment (johnallen3d) [inline projects/cuelang.org/package.yml:31]: This is slightly more than the bare minimum (e.g. `cue version`). Happy to add more test if you feel it's appropriate.

**Comment by johnallen3d** [comment]
Thanks for the review @jhheider! I've made the requested changes.

**Comment by mxcl** [comment]
thanks @johnallen3d. Sorry for the hassles in building. I am working on making this simpler. The instructions are a little out of date also since cloning `cli` is no longer required.

**Comment by johnallen3d** [comment]
No problem at all @mxcl, I figured reporting the rough edges might help provide some external user perspective.

**Comment by mxcl** [comment]
Thank you 🙌🏻

**Pull request merged**